iphone/Classes/KrollBridge.m

```diff
@@ -1009,37 +988,63 @@ - (TiModule *)loadJavascriptText:(NSString *)data fromFile:(NSString *)filename
 	return module;
 }
 
-- (TiModule *)loadAsFile:(NSString *)path withContext:(KrollContext *)kroll
+- (KrollWrapper *)cachedLoadAsFile:(NSString *)path asJSON:(bool)json withContext:(KrollContext *)kroll
 {
-	// 1. If X is a file, load X as JavaScript text.  STOP
-	NSString *filename = path;
-	NSString *data = [self loadFile:filename];
+	// check cache first
+	if (modules != nil) {
+		KrollWrapper *module = [modules objectForKey:path];
+		if (module != nil) {
+			NSLog(@"[WARN][MODULE CACHE HIT] : %@", path);
+			return module;
+		}
+		NSLog(@"[WARN][MODULE CACHE MISS] : %@", path);
```

Comment:
I left these in intentionally for testing this PR. I do plan to remove them before we merge. Just thought it'd be easier for you'QE to have some transparency about what's going on under the hood to validate behavior.